src/bun.js/bindings/NodeTLS.cpp

```diff
@@ -72,6 +73,48 @@ JSC_DEFINE_HOST_FUNCTION(getExtraCACertificates, (JSC::JSGlobalObject * globalOb
     RELEASE_AND_RETURN(scope, JSValue::encode(JSC::objectConstructorFreeze(globalObject, rootCertificates)));
 }
 
+JSC_DEFINE_HOST_FUNCTION(getSystemCACertificates, (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    auto scope = DECLARE_THROW_SCOPE(globalObject->vm());
+    VM& vm = globalObject->vm();
+
+    STACK_OF(X509)* root_system_cert_instances = us_get_root_system_cert_instances();
+
+    auto size = sk_X509_num(root_system_cert_instances);
+    if (size < 0) size = 0; // root_system_cert_instances is nullptr
+
+    auto rootCertificates = JSC::JSArray::create(vm, globalObject->arrayStructureForIndexingTypeDuringAllocation(JSC::ArrayWithContiguous), size);
```

Comment:
This code should be rewritten to use a MarkedArgumentsBuffer, append to the marked arguments buffer, and then at the end use JSC::objectConstructorFreeze(JSC::constructArray on that marked arguments buffer